prow/spyglass/spyglass.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package spyglass creates views for Prow job artifacts.
+package spyglass
+
+import (
+	"errors"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/deck/jobs"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+var (
+	// These are artifacts that we guarantee will be created with every job.
+	// If they are found ina storage location, the stored ones will be used,
+	// otherwise we will try to fetch them from the job runtimeIn the future
+	// this might contain job metadata, podspec, etc.
+	guaranteedArtifacts = []string{"build-log.txt"}
```

Comment:
Not 100% sure that these artifacts are guaranteed to exist. We have jobs right now that do not use bootstrap and do not use the decorators, thus they do not have a build-log.txt

Reply:
A. Shouldn't we be uploading logs for those jobs then?
B. We gracefully handle missing logs for situations like prowjob no longer exists or job got TTL'd